app/javascript/components/Greeting.js

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchRandomGreeting } from '../redux/actions';
+
+function Greeting() {
+    const greeting = useSelector((state) => state.greeting);
+    const loading = useSelector((state) => state.loading);
+    const error = useSelector((state) => state.error);
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        console.log('Fetching random greeting...');
+        dispatch(fetchRandomGreeting());
+    }, [dispatch]);
+
+    console.log('Greeting:', greeting);
+    console.log('Loading:', loading);
+    console.log('Error:', error);
+
+    return (
+        <div>
+            {loading ? (
+                <p>Loading...</p>
+            ) : (
+                error ? (
+                    <p>
+                        Error:
+                        {error}
+                    </p>
+                ) : (
+                    <div>
+                        <h1>Random Greeting:</h1>
+                        <p>{greeting}</p>
+                    </div>
+                )
+            )}
```

Comment:
- [ ] Because you're not using the App.js component the Greeting.js component is not used either which in turn causes Redux to not be used in your application.